tests/testthat/test-qenv_eval_code.R

```diff
@@ -108,3 +108,47 @@ testthat::test_that("an error when calling eval_code returns a qenv.error object
   )
   testthat::expect_equal(q$message, "object 'w' not found \n when evaluating qenv code:\n z <- w * x")
 })
+
+testthat::test_that("a warning when calling eval_code returns a qenv object which has warnings", {
+  q <- eval_code(new_qenv(), quote("iris_data <- iris"))
+  q <- eval_code(q, quote("p <- hist(iris_data[, 'Sepal.Length'], ff = '')"))
+  testthat::expect_s4_class(q, "qenv")
+  testthat::expect_equal(
+    q@warnings,
+    c(
+      "",
+      paste0(
+        "\"ff\" is not a graphical parameter",
+        "\"ff\" is not a graphical parameter",
+        "\"ff\" is not a graphical parameter",
+        "\"ff\" is not a graphical parameter"
```

Comment:
I was confused on an initial read of this test as to why there are 4 of the same warning message. I wonder if a simpler example would work here? Maybe something like `q <- eval_code(new_qenv(), quote("warning('warning')"))`.

Reply:
@asbates I would leave it as it is now to make sure that any future change in the code still reports all the warnings rendered and not only a specific one.
For instance, this code:
```
iris_data <- iris
p <- hist(iris_data[, 'Sepal.Length'], ff = '')
```
renders four warnings:
```
Warning messages:
1: In plot.window(xlim, ylim, "", ...) : "ff" is not a graphical parameter
2: In title(main = main, sub = sub, xlab = xlab, ylab = ylab, ...) :
  "ff" is not a graphical parameter
3: In axis(1, ...) : "ff" is not a graphical parameter
4: In axis(2, ...) : "ff" is not a graphical parameter
```
and we optimally have to report the 4 of them to the user to not truncate any information.
Does this make sense to you @asbates ?

Reply:
@asbates can you update this? thanks

Reply:
@mhallal1 @nikolas-burkoff It makes more sense to me now. This is a test to ensure that multiple warnings from a single line of code are captured. What about updating the test description instead of a comment/another code line? Something that invokes the test purpose a little more clearly. I can take care of this.